Doc/library/email.utils.rst

```diff
@@ -58,13 +58,18 @@ of the new API.
    begins with angle brackets, they are stripped off.
 
 
-.. function:: parseaddr(address)
+.. function:: parseaddr(address, *, strict=True)
```

Comment:
> The incompatible change is obviously the fact the emails accepted previously are now rejected.

Yes.
If the default stayed `strict=False`, or to only *warn* by default on invalid addresses (for 2 releases), this would match the process in PEP 387. With `strict=True`, it seems that you'll need a SC exception for backwards incompatible behaviour -- and that can take time.

Or do you consider this a plain bug fix? In that case, why preserve the old behaviour?

Reply:
This change falls into the "unfortunately interesting" category of being unsure we're doing everything correctly so it provides an opt-out. If we have confidence in our change and test suite of valid and invalid email address coverage we could proceed without the behavior flag.

Do we have such confidence?

Reply:
I have 0% confience that this change make the parser fully compatible with all RFC. I have 100% confidence that this change rejects email addresses which were accepted before.

Sorry, as I wrote before, the whole parser should be rewritten from scratch to use grammar suggested by RFC, rather than the current design.

This change only make the situation "less bad".

Also it's common that when Python becomes stricter, someone comes and explain their valid use case relying on the old behaviour.